### PR TITLE
update deps, adopt v1 api

### PR DESCRIPTION
**Pull request by jzelinskie**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,17 +6,25 @@ require (
 	github.com/Azure/go-ansiterm v0.0.0-20210617225240-d185dfc1b5a1 // indirect
 	github.com/Microsoft/go-winio v0.5.0 // indirect
 	github.com/Nvveen/Gotty v0.0.0-20120604004816-cd527374f1e5 // indirect
-	github.com/authzed/authzed-go v0.2.0
+	github.com/authzed/authzed-go v0.3.0
```
Comment (ecordell) [inline go.mod:9]: 0.4.1 is the latest now

**Review by ecordell — COMMENTED** [review]
a flag bikeshed:

```
      --proxy-addr string                            address to listen on to serve proxy (default ":9090")
      --proxy-enabled                                enable proxy http server (default true)
      --proxy-cors-allowed-origins strings           allowed origins for CORS requests (default [*])
      --proxy-tls-cert-path string                   local path to the TLS certificate used to serve proxy
      --proxy-tls-key-path string                    local path to the TLS key used to serve proxy


      --proxy-check-permission string                permission to check
      --proxy-check-resource-id-query-param string   query parameter used as the Object ID to check
      --proxy-check-resource-type string             resource type to check
      --proxy-check-subject-relation string          optional subject relation to check
      --proxy-check-subject-type string              subject type to check
      --proxy-spicedb-endpoint string                address of the Authzed to use for checking (default "grpc.authzed.com:443")
      --proxy-spicedb-insecure                       connect to Authzed without TLS
      --proxy-spicedb-tls-cert-path string           path at which to find a certificate for authzed TLS
      --proxy-spicedb-token string                   authzed token to use for checking tenancy

      --proxy-upstream-prometheus-addr string        address of the upstream Prometheus
```

do we want the `proxy` prefix on the flags in the second block? they make sense to me on the first block (configuring the proxy endpoint itself) but the rest just seem like config for the whole program

File: main.go [hunk 2]
```diff
@@ -26,307 +28,192 @@ import (
...
+			Subject: &v1pb.SubjectReference{
+				Object: &v1pb.ObjectReference{
+					ObjectType: subjectType,
+					ObjectId:   strings.TrimPrefix(bearerToken, bearerTokenPrefix),
+				},
+				OptionalRelation: subjectRelation,
+			},
+		})
+		if err != nil {
+			http.Error(w, "upsteam failure", http.StatusServiceUnavailable)
```
Comment (josephschorr) [inline main.go:208]: Mind logging the error here?

**Comment by jzelinskie** [comment]
I treated the whole system as "proxy" in case at some point this program ends up running multiple types of servers. That's how we've done things with SpiceDB, so that it's a bit less confusing.

File: README.md [hunk 3]
```diff
@@ -46,16 +50,15 @@ The following command will run the proxy that checks the permissions against [au
...
-    --authzed-token tc_client_token_1234deadbeef  \
-    --authzed-subject-definition-path psystem/token \
-    --authzed-subject-relation ... \
-    --authzed-object-definition-path psystem/prometheus \
-    --authzed-permission viewer
+    --proxy-upstream-prometheus-addr http://localhost:9090 \
+    --proxy-spicedb-token tc_client_token_1234deadbeef  \
+    --proxy-check-resource-type psystem/prometheus \
+    --proxy-check-resource-id-query-param install \
+    --proxy-check-permission viewer
```
Comment (josephschorr) [inline README.md:57]: Make this `view`, since it is a permission

File: main_test.go [hunk 4]
```diff
@@ -71,19 +68,27 @@ func TestValidToken(t *testing.T) {
...
+		&v1.WriteRelationshipsRequest{
+			Updates: []*v1.RelationshipUpdate{
+				{
+					Operation: v1.RelationshipUpdate_OPERATION_CREATE,
+					Relationship: &v1.Relationship{
+						Resource: &v1.ObjectReference{
+							ObjectType: "test/dashboard",
+							ObjectId:   "foobar",
+						},
+						Relation: "viewer",
```
Comment (josephschorr) [inline main_test.go:82]: ditto

Comment (jzelinskie) [reply main_test.go:82]: This is actually writing the relationship. It's correct to be viewer -- the test is actually using view to check.

**Review by ecordell — APPROVED** [review]
LGTM

**Pull request merged**